src/pytest_mock_resources/container/base.py

```diff
@@ -1,55 +1,150 @@
+import contextlib
+import functools
+import json
+import pathlib
 import time
+from typing import Optional
 
-import docker
 import responses
 
+from pytest_mock_resources.config import get_env_config
+
+DEFAULT_RETRIES = 40
+DEFAULT_INTERVAL = 0.5
+
 
 class ContainerCheckFailed(Exception):
     """Unable to connect to a Container."""
 
 
-def get_container(config, ports, environment, check_fn):
+def retry(func=None, *, args=(), kwargs={}, retries=1, interval=DEFAULT_INTERVAL, on_exc=Exception):
+    while retries:
+        retries -= 1
+        try:
+            result = func(*args, **kwargs)
+        except on_exc:
+            if not retries:
+                raise
+            time.sleep(interval)
+        else:
+            return result
+
+
+def get_container(
+    pytestconfig,
+    config,
+    *,
+    check_fn,
+    ports=None,
+    environment=None,
+    retries=DEFAULT_RETRIES,
+    interval=DEFAULT_INTERVAL
+):
+    import docker
+    import docker.errors
```

Comment:
how come these are local imports?

Reply:
i'm trying to turn docker into an optional dependency that's included on use of the existing extras kinds.

Reply:
**perhaps** docker being optional isn't super interesting because sqlite is the exception more than the rule. **however** all docker-based fixtures need their own extras anyways (psycopg2, mysql conenctor, rabbit client, redis client, etc), so it's not really a UX loss to make it optional anyways.